### PR TITLE
fix(list-images): filter images by `environment=production`

### PR DESCRIPTION
**Pull request by fruch**
in order to minimize the noise from development images, we should list as offical released images only images labels with `environment=production`

### Testing

- [ ]

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Review by roydahan — APPROVED** [review]

**Comment by fruch** [comment]
it's currently a bit problematic, since this label doesn't existing on older releases (i.e. ones used for oracle in gemini tests)
@yaronkaikov can we apply this label to all live releases images  ?

**Comment by yaronkaikov** [comment]
> it's currently a bit problematic, since this label doesn't existing on older releases (i.e. ones used for oracle in gemini tests) @yaronkaikov can we apply this label to all live releases images ?

It's already applied on 6.1,6.2, 2024.1 and 2024.2

**Comment by fruch** [comment]
> > it's currently a bit problematic, since this label doesn't existing on older releases (i.e. ones used for oracle in gemini tests) @yaronkaikov can we apply this label to all live releases images ?
> 
> It's already applied on 6.1,6.2, 2024.1 and 2024.2

we are still using 2022.1 version as gemini oracle:
```
oracle_scylla_version: '2022.1.14'
```

so we'll need some backward compatibility, to move to using that tab

**Comment by yaronkaikov** [comment]
> > > it's currently a bit problematic, since this label doesn't existing on older releases (i.e. ones used for oracle in gemini tests) @yaronkaikov can we apply this label to all live releases images ?
> > 
> > 
> > It's already applied on 6.1,6.2, 2024.1 and 2024.2
> 
> we are still using 2022.1 version as gemini oracle:
> 
> ```
> oracle_scylla_version: '2022.1.14'
> ```
> 
> so we'll need some backward compatibility, to move to using that tab

Skip it :-), we are not going to release a new official AMI for that, i can add this tag manually on the latest one for main region or something if you can overcome it

**Comment by fruch** [comment]
> > > > it's currently a bit problematic, since this label doesn't existing on older releases (i.e. ones used for oracle in gemini tests) @yaronkaikov can we apply this label to all live releases images ?
> > > 
> > > 
> > > It's already applied on 6.1,6.2, 2024.1 and 2024.2
> > 
> > we are still using 2022.1 version as gemini oracle:
> > 
> > ```
> > oracle_scylla_version: '2022.1.14'
> > ```
> > 
> > so we'll need some backward compatibility, to move to using that tab
> 
> Skip it :-), we are not going to release a new official AMI for that, i can add this tag manually on the latest one for main region or something if you can overcome it
> 

I'll tag them myself, I found a way to do that quickly

File: sdcm/utils/common.py [hunk 1]
```diff
@@ -1132,14 +1132,14 @@ def filter_k8s_clusters_by_tags(tags_dict: dict, clusters: list[
 @lru_cache
 def get_scylla_ami_versions(region_name: str, arch: AwsArchType = 'x86_64', version: str = None) -> list[EC2Image]:
```
Comment (soyacz) [inline sdcm/utils/common.py:1133]: it's used by `get_latest_scylla_ami_release` which is not used anywhere, maybe remove this function along so it's not distracting?

**Review by soyacz — APPROVED** [review]
LGTM

File: sdcm/utils/common.py (hunk 1 above)
Comment (vponomaryov) [reply sdcm/utils/common.py:1133]:
```
$ git grep get_scylla_ami_versions
sdcm/sct_config.py:    get_scylla_ami_versions,
sdcm/sct_config.py:                            ami = get_scylla_ami_versions(version=scylla_version, region_name=region, arch=aws_arch)[0]
sdcm/sct_config.py:                            ami = get_scylla_ami_versions(version=oracle_scylla_version,
sdcm/utils/common.py:def get_scylla_ami_versions(region_name: str, arch: AwsArchType = 'x86_64', version: str = None) -> list[EC2Image]:
sdcm/utils/common.py:    for ami in get_scylla_ami_versions(region_name=region):
sdcm/utils/common.py:            for ami in get_scylla_ami_versions(region_name=region_name, arch=arch, version=version)]
unit_tests/test_scylla_yaml_builders.py:            with patch("sdcm.sct_config.get_scylla_ami_versions", return_value=[self.get_scylla_ami_version_output]), \

```

Comment (soyacz) [reply sdcm/utils/common.py:1133]: I meant `get_latest_scylla_ami_release` to be removed

**Pull request merged**